src/main/java/com/dotloop/loopit/LoopItController.java

```diff
@@ -54,51 +51,102 @@
     @Value("${dotloop.api.endpoint}")
     private String apiEndpoint;
 
-    @RequestMapping("/loopit")
+    private String username = "api user";
+
+    private Request attachHeaders(Request request) {
+        Token token = TokenStore.get(username);
+        request.addHeader("Authorization", "Bearer " + token.getAccessToken())
+                .addHeader("Content-type", "application/json");
+
+        return request;
+    }
+
+    private Request attachBody(Request request,  String bodyData) {
+        request.bodyString(bodyData, ContentType.APPLICATION_JSON);
+
+        return request;
+    }
+
+    private String makeRequest(String type, String params, String bodyData) throws Exception {
+        try {
+            params = params.isEmpty() ? "" : params;
+
+            Request request = attachHeaders(Request.Get(getApiBaseUrl() + params));
+
+            if (type.equalsIgnoreCase("post")) {
+                request = attachBody(attachHeaders(Request.Post(getApiBaseUrl() + params)), bodyData);
+            }
+
+            HttpResponse response = request.execute().returnResponse();
+
+            String payload = IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+            logger.debug("Response: " + payload);
+            if (response.getStatusLine().getStatusCode() >= 300) {
+                throw new HttpResponseException(response.getStatusLine().getStatusCode(),
+                        response.getStatusLine().getReasonPhrase());
+            }
+
+            return payload;
+
+        } catch (Exception e) {
+            logger.error("Something unexpected happened: " + e.getMessage());
+            TokenStore.delete(username); // fixme - needed only for 401
+            throw new RuntimeException(e); // todo handle error - token revoked, etc
+        }
+
+    }
+
+    private String makeRequest(String type, String params) throws Exception {
+        return makeRequest(type, params, "");
+    }
+
+    @RequestMapping(value = "/loopit", method = RequestMethod.POST)
     @ResponseBody
-    public String loopIt() {
-        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+    public String loopIt(
+            @RequestParam(value = "profile_id", required = true) String profile_id,
+            @RequestBody Loop loop) throws Exception {
+
+        Token token = TokenStore.get(username);
 
-        Token token = TokenStore.get(user.getUsername());
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
@sweetcoco can you make this a member variable?  In many cases the object mapper will have customized serialization settings that need to be shared.  Not really the case here, but good practice.